service/controller/v13/resource/node/error.go

```diff
@@ -0,0 +1,74 @@
+package node
+
+import (
+	"regexp"
+
+	"github.com/giantswarm/microerror"
+)
+
+var invalidConfigError = microerror.New("invalid config")
+
+// IsInvalidConfig asserts invalidConfigError.
+func IsInvalidConfig(err error) bool {
+	return microerror.Cause(err) == invalidConfigError
+}
+
+const (
+	// guestDNSNotReadyPattern is a regular expression representing DNS errors for
+	// the guest API domain. Also see the following match example.
+	//
+	//     https://play.golang.org/p/ipBkwqlc4Td
+	//
+	guestDNSNotReadyPattern = "dial tcp: lookup .* on .*:53: no such host"
+
+	// guestEOFPattern is a regular expression representing EOF errors for the
+	// guest API domain. Also see the following match example.
+	//
+	//     https://play.golang.org/p/L6f4ItJLufv
+	//
+	guestEOFPattern = `Get https://api\..*/api/v1/nodes: EOF`
+
+	// guestTransientInvalidCertificatePattern regular expression defines the kind
+	// of transient errors related to certificates returned while the guest API is
+	// not fully up. Also see the following match example.
+	//
+	//     https://play.golang.org/p/iiYvBhPOg4f
+	//
+	guestTransientInvalidCertificatePattern = `[Get|Post] https://api\..*: x509: certificate is valid for ingress.local, not api\..*`
+)
+
+var (
+	guestDNSNotReadyRegexp                 = regexp.MustCompile(guestDNSNotReadyPattern)
+	guestEOFRegexp                         = regexp.MustCompile(guestEOFPattern)
+	guestTransientInvalidCertificateRegexp = regexp.MustCompile(guestTransientInvalidCertificatePattern)
+)
+
+var guestAPINotAvailableError = microerror.New("guest API not available")
+
+// IsGuestAPINotAvailable asserts guestAPINotAvailableError.
+func IsGuestAPINotAvailable(err error) bool {
```

Comment:
I took this from `cluster-operator` and extended it with the EOF checks. We should separate this and use the same mechanism everywhere. I can imagine the EOF errors are a thing in other places as well.

Reply:
@fgimenez @rossf7 WDYT? Where to put it? `giantswarm/errors`?

Reply:
@xh3b4sd Yes we need this logic in multiple places including helmclient. I've already created the repo but I got distracted by other things. I'll pick this up next.

https://github.com/giantswarm/errors